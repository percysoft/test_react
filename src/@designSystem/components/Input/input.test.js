import React from 'react';
import '@testing-library/jest-dom';
import { InputContainer } from './index';
import { shallow } from 'enzyme';

describe('Test Input Component', () => {
  test('show component', () => {
    const wrapper  = shallow (<InputContainer />)
    expect(wrapper).toMatchSnapshot();
  })

  test('Show mount only  min and max', () => {
    let min_amount = 1;
    let max_amount= 50;
    let message_test = `Mínimo S/${min_amount} - Máximo S/ ${max_amount}`;
    const wrapper  = shallow (<InputContainer min_amount={min_amount} max_amount={max_amount}/>);
    const wrapper_text = wrapper.find('div').text().trim();
    expect(wrapper_text).toBe(message_test);
  }) 
})