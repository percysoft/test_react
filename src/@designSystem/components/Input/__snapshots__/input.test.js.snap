// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Input Component show component 1`] = `
<div>
  <WithStyles(ForwardRef(TextField))
    InputProps={
      Object {
        "inputComponent": [Function],
      }
    }
    fullWidth={true}
    id="formatted-numberformat-input"
    inputProps={
      Object {
        "data-testid": "content-input",
        "id": "content-input",
      }
    }
    name="numberformat"
    onChange={[Function]}
    value={100}
  />
  <styled.div
    bold="300"
    size="15"
  >
    Mínimo S/1 - Máximo S/ 50
  </styled.div>
</div>
`;
